prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -833,6 +849,44 @@ class Hl7Serializer(val metadata: Metadata): Logging {
         }
     }
 
+    /**
+     * Decode an XTN (e.g. phone number) component of an HL7 message.
+     * @param terser the HL7 terser
+     * @param element the element to decode
+     * @return the phone number or empty string
+     */
+    internal fun decodeXTNPhoneNumber(terser: Terser, element: Element): String {
+        var decodedPhoneNumber = ""
+
+        // Telephone is of XTN type in HL7
+        var deprecatedPhoneNumber = ""
+        var countryCode = ""
+        var areaCode = ""
+        var localNumber = ""
+        var equipType = ""
+
+        // Note: Let the calling function catch any HL7 exception
+        deprecatedPhoneNumber = terser.get("/.${element.hl7Field}-1") ?: ""
+        equipType = terser.get("/.${element.hl7Field}-3") ?: ""
+        countryCode = terser.get("/.${element.hl7Field}-5") ?: ""
+        areaCode = terser.get("/.${element.hl7Field}-6") ?: ""
+        localNumber = terser.get("/.${element.hl7Field}-7") ?: ""
+
+        // TODO How to handle extensions
+
+        // Make sure we only grab a phone number if the type is specified
+        if(equipType.isEmpty() || equipType == "PH") {
+            // Use the deprecated phone number if nothing else is available.
+            if(countryCode.isBlank() && areaCode.isBlank() && localNumber.isBlank()) {
+                decodedPhoneNumber = deprecatedPhoneNumber
+            } else {
+                decodedPhoneNumber = "$countryCode$areaCode$localNumber"
+            }
+        }
+
+        return decodedPhoneNumber
+    }
```

Comment:
I need to double check, but should the value be stripped of formatting if it's in the `deprecatedPhoneNumber` field? Also, should the number be normalized before being set on the element?

Reply:
You are right.  One sample HL7 data had a formatted phone number.  I stripped all formatting from the deprecated phone number